src/odemis/acq/millmng.py

```diff
@@ -0,0 +1,519 @@
+        """
+        data = []
+        # check if cancellation happened while the acquiring future is working
+        with self._future._task_lock:
+            if self._future._task_state == CANCELLED:
+                raise CancelledError()
+        try:
+            self._future.running_subf = acquire(self.streams)
+            data, exp = self._future.running_subf.result()
+            if exp:
+                logging.warning(
+                    f"Acquisition for feature {site.name.value} partially failed: {exp}")
+        except Exception as exp:
+            logging.debug(f"Acquisition for feature {site.name.value} failed: {exp}")
+
+        # Check on the acquired data
+        if not data:
+            logging.warning("The acquired data array in stream %s for feature %s is empty", self.streams,
+                            site.name)
+        else:
+            # Convert the data to StaticStreams, and add them to the feature
+            new_streams = dataio.data_to_static_streams(data)
+            site.streams.value.extend(new_streams)
+            logging.info("The acquisition for stream %s for feature %s is done", self.streams, site.name)
+
+    def run(self):
+        """
+        The main function of the task class, which will be called by the future asynchronously
+        """
+        self._future._task_state = RUNNING
+
+        try:
+            with self._future._task_lock:
+                if self._future._task_state == CANCELLED:
+                    raise CancelledError()
+            # set the FIB mode with the retracted state of the objective
+            # get the metadata to retract the objective
+            align_md = self._aligner.getMetadata()
+            deactive_pos = align_md[model.MD_FAV_POS_DEACTIVE]
+
+            # create a future and update the appropriate controls after it is called
+            # TODO: use move.cryoSwitchSamplePosition(MILLING)
+            logging.debug(f"Retracting the objective to set the imaging in FIB mode")
+            self._future.running_subf = self._aligner.moveAbs(deactive_pos)
+
+            try:
+                self._future.running_subf.result()
+            except Exception:
+                raise
+
+            actual_time_per_site = None
+
+            for site_idx, site in enumerate(self.sites):
+                # Update progress on the milling sites left
+                start_time = time.time()
+                remaining_t = self.estimate_milling_time(site_idx, actual_time_per_site)
+                self._future.set_end_time(time.time() + remaining_t)
+
+                # For one given site, move the stage at the site and
+                # do milling with all milling settings for the given site
+                self._mill_all_settings(site)
+                # If timeout exception is raised during stage movement, continue milling the next site
+                if Exception == TimeoutError:
+                    continue
```

Comment:
That code makes no sense. `Exception` is a class. `TimeoutError` is a class. They are never equal.
I don't have a really nice way to handle the special case that the stage failed to move, so we just skipped the milling. Maybe make mill_all_settings() return True/False. Or crease a special exception, like "MoveError", to indicate this specific failure type (and use a try/except block).

Reply:
Adding a special class MoveError(Exception) at line 94